lib/manifest.ex

```diff
@@ -77,6 +92,15 @@ defmodule Manifest do
   """
 
   @spec add_step(t(), Step.t()) :: t()
+  def add_step(manifest, step(operation: operation) = step) do
+    if MapSet.member?(manifest.operations, operation),
```

Comment:
do we need to add something like this to `merge/2`? since the steps there are added in a different way